internal/backend/sessions/sessionworkflows/syscalls.go

```diff
@@ -23,32 +24,76 @@ const (
 	unsubscribeOp = "unsubscribe"
 )
 
-func (w *sessionWorkflow) syscall(ctx context.Context, args []sdktypes.Value, kwargs map[string]sdktypes.Value) (sdktypes.Value, error) {
-	if len(args) == 0 {
-		return sdktypes.InvalidValue, fmt.Errorf("expecting syscall operation name as first argument")
-	}
+func (w *sessionWorkflow) syscalls() *sdkservices.RunSyscalls {
+	call := func(ctx context.Context, name string, args []sdktypes.Value, kwargs map[string]sdktypes.Value) (sdktypes.Value, error) {
+		v := w.akModule.Values()[name]
 
-	var op string
+		wctx := sessioncontext.GetWorkflowContext(ctx)
+		if wctx == nil {
+			return sdktypes.InvalidValue, errors.New("missing workflow context")
+		}
 
-	if err := sdktypes.DefaultValueWrapper.UnwrapInto(&op, args[0]); err != nil {
-		return sdktypes.InvalidValue, err
+		return w.call(wctx, v, args, kwargs)
 	}
 
-	args = args[1:]
-
-	switch op {
-	case sleepOp:
-		return w.sleep(ctx, args, kwargs)
-	case startOp:
-		return w.start(ctx, args, kwargs)
-	case subscribeOp:
-		return w.subscribe(ctx, args, kwargs)
-	case nextEventOp:
-		return w.nextEvent(ctx, args, kwargs)
-	case unsubscribeOp:
-		return w.unsubscribe(ctx, args, kwargs)
-	default:
-		return sdktypes.InvalidValue, fmt.Errorf("unknown op %q", op)
+	return &sdkservices.RunSyscalls{
```

Comment:
Why is `RunSyscalls` a struct and not an interface? I don't understand the purpose of `RunSyscalls` and how it helps the code.